embassy-stm32/src/sai/mod.rs

```diff
@@ -531,10 +531,13 @@ pub struct SubBlock<'d, T: Instance, C: Channel, W: word::Word> {
 pub struct SubBlockA {}
 pub struct SubBlockB {}
 
+pub struct SubBlockAPeripheral<'d, T>(PeripheralRef<'d, T>);
```

Comment:
could you take these unrelated changes out? thanks!

Reply:
This aren't unrelated changes.
While the SAI is only one Peripheral Block, it contains two complety independent subblocks.
Currently it is possible to create a subblock directly, or frist create a SAI peripheral, which just copies the PeripheralRef, so we can initalise both subblocks.
Question, where do we call enable_and_reset()?

This changes force the user to create a SAI first, which enables clock and reset, and then use the take_sub_block to get a SubBlockXPeripheral to initalise the Subblock

Reply:
ahh got it, makes sense, thank you!